File: src/main/scala/resonantinduction/mechanical/mech/process/mixer/MixerNode.java
package resonantinduction.mechanical.mech.process.mixer;

import net.minecraftforge.common.util.ForgeDirection;
import resonantinduction.mechanical.mech.MechanicalNode;
import universalelectricity.api.core.grid.INodeProvider;

public class MixerNode extends MechanicalNode
{

    public MixerNode(INodeProvider parent)
    {
        super(parent);
        maxDeltaAngle = Math.toRadians(45);
        sharePower = false;
    }

    @Override
    public boolean canConnect(ForgeDirection direction)
    {
        return direction == ForgeDirection.DOWN || direction == ForgeDirection.UP;
    }

    @Override
    public boolean inverseRotation(ForgeDirection dir)
    {
        return dir == ForgeDirection.DOWN;
    }

}

File: src/main/scala/resonantinduction/mechanical/mech/process/mixer/MixerNode.scala
package resonantinduction.mechanical.mech.process.mixer

import net.minecraftforge.common.util.ForgeDirection
import resonant.api.tile.INodeProvider
import resonantinduction.mechanical.mech.grid.NodeMechanical

/**
 * Node designed just for the Mixer to use
 * @param parent - instance of TileMixer that will host this node, should never be null
 */
class MixerNode(parent: INodeProvider) extends NodeMechanical(parent)
{
    override def canConnect(direction: ForgeDirection): Boolean =
    {
        return direction == ForgeDirection.DOWN || direction == ForgeDirection.UP
    }

    /*
    override def inverseRotation(dir: ForgeDirection): Boolean =
    {
        return dir == ForgeDirection.DOWN
    }*/
}